File: src/main/scala/org/openapitools/models/FullOutletDTO.scala
package org.openapitools.models

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import ympa_scala_finch_server._
import org.openapitools.models.OutletAddressDTO
import org.openapitools.models.OutletDeliveryRuleDTO
import org.openapitools.models.OutletStatusType
import org.openapitools.models.OutletType
import org.openapitools.models.OutletVisibilityType
import org.openapitools.models.OutletWorkingScheduleDTO
import org.openapitools.models.RegionDTO
import scala.collection.immutable.Seq

/**
 * Информация о точке продаж.
 * @param name Название точки продаж. 
 * @param _type 
 * @param coords Координаты точки продаж.  Формат: долгота, широта. Разделители: запятая и / или пробел. Например, `20.4522144, 54.7104264`.  Если параметр не передан, координаты будут определены по значениям параметров, вложенных в `address`. 
 * @param isMain Признак основной точки продаж.  Возможные значения:  * `false` — неосновная точка продаж. * `true` — основная точка продаж. 
 * @param shopOutletCode Идентификатор точки продаж, присвоенный магазином.
 * @param visibility 
 * @param address 
 * @param phones Номера телефонов точки продаж. Передавайте в формате: `+7 (999) 999-99-99`. 
 * @param workingSchedule 
 * @param deliveryRules Информация об условиях доставки для данной точки продаж.  Обязательный параметр, если параметр `type=DEPOT` или `type=MIXED`. 
 * @param storagePeriod Срок хранения заказа в собственном пункте выдачи заказов. Считается в днях.
 * @param id Идентификатор точки продаж, присвоенный Маркетом.
 * @param status 
 * @param region 
 * @param shopOutletId Идентификатор точки продаж, заданный магазином.
 * @param workingTime Рабочее время.
 * @param moderationReason Статус модерации.
 */
case class FullOutletDTO(name: String,
                _type: OutletType,
                coords: Option[String],
                isMain: Option[Boolean],
                shopOutletCode: Option[String],
                visibility: Option[OutletVisibilityType],
                address: OutletAddressDTO,
                phones: Seq[String],
                workingSchedule: OutletWorkingScheduleDTO,
                deliveryRules: Option[Seq[OutletDeliveryRuleDTO]],
                storagePeriod: Option[Long],
                id: Option[Long],
                status: Option[OutletStatusType],
                region: Option[RegionDTO],
                shopOutletId: Option[String],
                workingTime: Option[String],
                moderationReason: Option[String]
                )

object FullOutletDTO {
    /**
     * Creates the codec for converting FullOutletDTO from and to JSON.
     */
    implicit val decoder: Decoder[FullOutletDTO] = deriveDecoder
    implicit val encoder: ObjectEncoder[FullOutletDTO] = deriveEncoder
}

File: lib/app/Models/FullOutletDTO.php
<?php
/**
 * FullOutletDTO
 */
namespace app\Models;

/**
 * FullOutletDTO
 * @description Информация о точке продаж.
 */
class FullOutletDTO {

    /** @var string $name Название точки продаж.*/
    public $name = "";

    /** @var string $type */
    public $type = "";

    /** @var string $coords Координаты точки продаж.  Формат: долгота, широта. Разделители: запятая и / или пробел. Например, &#x60;20.4522144, 54.7104264&#x60;.  Если параметр не передан, координаты будут определены по значениям параметров, вложенных в &#x60;address&#x60;.*/
    public $coords = "";

    /** @var bool $isMain Признак основной точки продаж.  Возможные значения:  * &#x60;false&#x60; — неосновная точка продаж. * &#x60;true&#x60; — основная точка продаж.*/
    public $isMain = false;

    /** @var string $shopOutletCode Идентификатор точки продаж, присвоенный магазином.*/
    public $shopOutletCode = "";

    /** @var string $visibility */
    public $visibility = "";

    /** @var \app\Models\OutletAddressDTO $address */
    public $address;

    /** @var string[] $phones Номера телефонов точки продаж. Передавайте в формате: &#x60;+7 (999) 999-99-99&#x60;.*/
    public $phones = [];

    /** @var \app\Models\OutletWorkingScheduleDTO $workingSchedule */
    public $workingSchedule;

    /** @var \app\Models\OutletDeliveryRuleDTO[]|null $deliveryRules Информация об условиях доставки для данной точки продаж.  Обязательный параметр, если параметр &#x60;type&#x3D;DEPOT&#x60; или &#x60;type&#x3D;MIXED&#x60;.*/
    public $deliveryRules = null;

    /** @var int $storagePeriod Срок хранения заказа в собственном пункте выдачи заказов. Считается в днях.*/
    public $storagePeriod = 0;

    /** @var int $id Идентификатор точки продаж, присвоенный Маркетом.*/
    public $id = 0;

    /** @var string $status */
    public $status = "";

    /** @var \app\Models\RegionDTO $region */
    public $region;

    /** @var string $shopOutletId Идентификатор точки продаж, заданный магазином.*/
    public $shopOutletId = "";

    /** @var string $workingTime Рабочее время.*/
    public $workingTime = "";

    /** @var string $moderationReason Статус модерации.*/
    public $moderationReason = "";

}
